Fix for URLSession redirect does not inherit request timeout value[SR…

saiHemak opened the pull request:
…-2682]

bubski commented:
@saiHemak does this maybe fix problems with `TestURLSession.test_httpRedirection`?

pushkarnk commented:
@saiHemak #1062 changed TestURLSession to using a single server. We'd need to change the test here accordingly.

xwu commented on Foundation/NSURLSession/http/EasyHandle.swift:
```diff
@@ -240,9 +240,18 @@ extension _EasyHandle {
-   }
+    }
+
+    func getTimetaken() -> Double {
```
Spelling: should be `getTimeTaken()`.

xwu commented on Foundation/NSURLSession/http/EasyHandle.swift:
```diff
@@ -240,9 +240,18 @@ extension _EasyHandle {
+    }
+
+    func getTimetaken() -> Double {
+        let timeTaken = UnsafeMutablePointer<Double>.allocate(capacity: 8)
```
This is not correct. You are allocating 8 instances of type `Double`. The body of this function should be:

```swift
var timeSpent = Double()
CFURLSession_easy_getinfo_double(rawHandle, CFURLSessionInfoTOTAL_TIME,  &timeSpent)
return timeSpent
```

Since this is used only for subtracting from the timeout interval, it might be worthwhile to return `timeTaken / 1000` and to rename this function `getTimeoutIntervalSpent()`.

pushkarnk commented:
@swift-ci please test

pushkarnk commented:
@swift-ci please test

saiHemak commented:
@ bubski  I could see the test case``` TestURLSession.test_httpRedirection``` is passing successfully  even without this change. .Can you please elaborate about the issue you are facing  ?
@pushkarnk ,@xwu : Thanks for the review.I have addressed your review comments.

bubski commented:
@saiHemak I pulled newest changes and the problem went away.

pushkarnk commented:
Thanks @saiHemak 

@swift-ci test and merge

pushkarnk commented:
@swift-ci test and merge
